registry/remote/repository.go

```diff
@@ -483,16 +483,14 @@ func (r *Repository) Referrers(ctx context.Context, desc ocispec.Descriptor, art
 
 	// The referrers state is unknown.
 	if err != nil {
-		var errResp *errcode.ErrorResponse
-		if !errors.As(err, &errResp) || errResp.StatusCode != http.StatusNotFound {
+		if !errors.Is(err, errdef.ErrUnsupported) {
 			return err
 		}
-		if errutil.IsErrorCode(errResp, errcode.ErrorCodeNameUnknown) {
+		if errutil.IsErrorCode(err, errcode.ErrorCodeNameUnknown) {
 			// The repository is not found, no fallback.
 			return err
 		}
```

Comment:
How about:
If we move the check for the repository not found case to `referrersByAPI`, we can do:
```go
	if err != nil {
		if errors.Is(err, errdef.ErrUnsupported) {
			// Referrers API is not supported, fallback to referrers tag schema.
			r.SetReferrersCapability(false)
			return r.referrersByTagSchema(ctx, desc, artifactType, fn)
		}
		return err
	}
```